src/base/NekRSProblem.C

```diff
@@ -62,8 +62,43 @@ NekRSProblem::NekRSProblem(const InputParameters & params)
   : NekRSProblemBase(params),
     _serialized_solution(NumericVector<Number>::build(_communicator).release()),
     _moving_mesh(getParam<bool>("moving_mesh")),
-    _has_heat_source(getParam<bool>("has_heat_source"))
+    _has_heat_source(getParam<bool>("has_heat_source")),
+    _usrwrk_indices(MultiMooseEnum("flux heat_source x_displacement y_displacement z_displacement unused"))
 {
+  // Determine an appropriate default usrwrk indexing; the ordering will always be
+  //   0: flux             (if _boundary is true)
+  //   1: heat_source      (if _volume is true and _has_heat_source is true)
+  //   2: x_displacement   (if _moving_mesh is true)
+  //   3: y_displacement   (if _moving_mesh is true)
+  //   4: z_displacement   (if _movign_mesh is true)
+  //
+  // The most we will do is skip allocating terms at the end of this ordering if we
+  // don't need them. We never change the ordering of "earlier" terms.
+  std::vector<std::string> indices =
+    {"flux", "heat_source", "x_displacement", "y_displacement", "z_displacement"};
+
+  // progressively erase terms from the back if we don't need them
+  if (!_moving_mesh)
```

Comment:
Do we want to support moving_mesh without CHT? E.g. for tensor mechanics + ALE solver in nek. In that case, this needs to be split up into separate if conditions.